### PR TITLE
Fix golint failures of test/e2e/storage/vsphere

### PR DESCRIPTION
**Pull request by Jarvis-Zhou**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Ref #68026

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/storage/vsphere/config.go [hunk 1]
```diff
@@ -98,7 +98,8 @@ func GetVSphereInstances() (map[string]*VSphere, error) {
 
 func getConfig() (*ConfigFile, error) {
 	if confFileLocation == "" {
-		return nil, fmt.Errorf("Env variable 'VSPHERE_CONF_FILE' is not set.")
+		msg := "Env variable 'VSPHERE_CONF_FILE' is not set"
+		return nil, errors.New(msg)
```
Comment (misterikkit) [inline test/e2e/storage/vsphere/config.go]: What is the reason for splitting this into two lines? Is there a specific lint error that recommends this?

File: test/e2e/storage/vsphere/nodemapper.go [hunk 2]
```diff
@@ -33,9 +33,11 @@ import (
 	neturl "net/url"
 )
 
+// NodeMapper holds method for node info map
```
Comment (misterikkit) [inline test/e2e/storage/vsphere/nodemapper.go]: nit: I think this comment could explain a bit more about what `NodeMapper` does. Maybe something like, `NodeMapper holds detailed info about nodes and datastore zones.`

File: test/e2e/storage/vsphere/nodemapper.go [hunk 3]
```diff
@@ -33,9 +33,11 @@ import (
...
 type NodeMapper struct {
 }
```
Comment (misterikkit) [inline test/e2e/storage/vsphere/nodemapper.go]: So this is probably not a golint error, but this type stores its state in global variables `nameToNodeInfo` and `vcToZoneDatastoresMap`. Those should really be inside this struct. Let me know if you think that should be done in a separate PR.

File: test/e2e/storage/vsphere/vsphere_stress.go [hunk 4]
```diff
@@ -114,21 +114,21 @@ var _ = utils.SIGDescribe("vsphere cloud provider stress [Feature:vsphere]", fun
...
+			go PerformVolumeLifeCycleInParallel(f, client, namespace, instanceID, scArrays[instanceCount%len(scArrays)], iterations, &wg)
 		}
 		wg.Wait()
 	})
 
 })
 
-// goroutine to perform volume lifecycle operations in parallel
-func PerformVolumeLifeCycleInParallel(f *framework.Framework, client clientset.Interface, namespace string, instanceId string, sc *storagev1.StorageClass, iterations int, wg *sync.WaitGroup) {
+// PerformVolumeLifeCycleInParallel goroutines to perform volume lifecycle operations in parallel
```
Comment (misterikkit) [inline test/e2e/storage/vsphere/vsphere_stress.go]: nit: this function is only in a goroutine if a caller adds `go` in front. In actuality, this function doesn't do anything in parallel. It would be more accurate to say that this function performs a complete volume lifecycle N times, serially. (And I would love it if we could rename the function.)

**Review by misterikkit — COMMENTED** [review]
Thanks for doing this cleanup! I have a few small comments.

File: test/e2e/storage/vsphere/config.go (hunk 1 above)
Comment (Jarvis-Zhou) [reply test/e2e/storage/vsphere/config.go]: Yes, golint recommends this

File: test/e2e/storage/vsphere/nodemapper.go (hunk 3 above)
Comment (Jarvis-Zhou) [reply test/e2e/storage/vsphere/nodemapper.go]: I think this should be done in a separate PR

File: test/e2e/storage/vsphere/nodemapper.go (hunk 2 above)
Comment (Jarvis-Zhou) [reply test/e2e/storage/vsphere/nodemapper.go]: I think this show be modified in a separate PR which move `nameToNodeInfo` and `vcToZoneDatastoresMap` into this Object.

File: test/e2e/storage/vsphere/vsphere_stress.go (hunk 4 above)
Comment (Jarvis-Zhou) [reply test/e2e/storage/vsphere/vsphere_stress.go]: OK. Should I add the modified lines into a new commit or amend into the commit before ?

Comment (Jarvis-Zhou) [reply test/e2e/storage/vsphere/vsphere_stress.go]: fixed

File: test/e2e/storage/vsphere/config.go (hunk 1 above)
Comment (Jarvis-Zhou) [reply test/e2e/storage/vsphere/config.go]: fixed

File: test/e2e/storage/vsphere/nodemapper.go (hunk 2 above)
Comment (misterikkit) [reply test/e2e/storage/vsphere/nodemapper.go]: I figured since this is a brand new comment, we could put a little more detail into it.

File: test/e2e/storage/vsphere/vsphere_stress.go [hunk 5]
```diff
@@ -114,21 +114,21 @@ var _ = utils.SIGDescribe("vsphere cloud provider stress [Feature:vsphere]", fun
...
 		}
 		wg.Wait()
 	})
 
 })
 
-// goroutine to perform volume lifecycle operations in parallel
-func PerformVolumeLifeCycleInParallel(f *framework.Framework, client clientset.Interface, namespace string, instanceId string, sc *storagev1.StorageClass, iterations int, wg *sync.WaitGroup) {
+// PerformVolumeLifeCycle performs volume lifecycle operations N times, serially.
+func PerformVolumeLifeCycle(f *framework.Framework, client clientset.Interface, namespace string, instanceID string, sc *storagev1.StorageClass, iterations int, wg *sync.WaitGroup) {
```
Comment (misterikkit) [inline test/e2e/storage/vsphere/vsphere_stress.go]: Thanks for renaming this!

**Review by misterikkit — COMMENTED** [review]
/lgtm

There is always more cleanup to do beyond just fixing lint errors, so further improvements are welcomed.

File: test/e2e/storage/vsphere/vsphere_stress.go (hunk 4 above)
Comment (misterikkit) [reply test/e2e/storage/vsphere/vsphere_stress.go]: I prefer git amend rather than adding commits, because at the end of the day, we have a git log and not a github pull request log.

File: test/e2e/storage/vsphere/nodemapper.go (hunk 2 above)
Comment (Jarvis-Zhou) [reply test/e2e/storage/vsphere/nodemapper.go]: In the next PR, I will move the two objects in NodeMapper and modify this comment. Is this OK?

Comment (misterikkit) [reply test/e2e/storage/vsphere/nodemapper.go]: Yes, that would be appreciated. :smile:

**Comment by jiaweizhou** [comment]
@misterikkit do you know anyone else can review and approve this pr?  I can not contact with verult and zmerlynn.

**Comment by Jarvis-Zhou** [comment]
@mikedanese @msau42 can you help review this pr?

**Comment by lavalamp** [comment]
/approve

for hack/

Sorry, didn't see this until now

**Comment by misterikkit** [comment]
Sorry this didn't get more prompt attention. Looks like you need to rebase.
